akka-stream/src/main/scala/akka/stream/SystemMaterializer.scala

```diff
@@ -27,9 +36,51 @@ object SystemMaterializer extends ExtensionId[SystemMaterializer] with Extension
 }
 
 final class SystemMaterializer(system: ExtendedActorSystem) extends Extension {
+  private val systemMaterializerPromise = Promise[Materializer]()
+
+  // load these here so we can share the same instance across materializer guardian and other uses
+  /**
+   * INTERNAL API
+   */
+  @InternalApi @silent("deprecated")
+  private[akka] val materializerSettings = ActorMaterializerSettings(system)
+
+  private implicit val materializerTimeout: Timeout =
+    system.settings.config.getDuration("akka.stream.materializer.creation-timeout").asScala
+
+  @InternalApi @silent("deprecated")
+  private val materializerGuardian = system.systemActorOf(
+    MaterializerGuardian
+      .props(systemMaterializerPromise, materializerSettings)
+      .withDispatcher(materializerSettings.dispatcher)
+      .withDeploy(Deploy.local),
+    "Materializers")
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  private[akka] def createAdditionalSystemMaterializer(): Materializer = {
+    val started =
+      (materializerGuardian ? MaterializerGuardian.StartMaterializer).mapTo[MaterializerGuardian.MaterializerStarted]
+    Await.result(started, materializerTimeout.duration).materializer
```

Comment:
So to keep API/not having to deal with a future we trade the potential per-graph-stage-Await.result to one-per- system wide materializer